discovery_engine/lib/src/domain/engine/engine.dart

```diff
@@ -26,12 +35,39 @@ abstract class Engine {
   /// Serializes the state of the [Engine] state.
   Future<Uint8List> serialize();
 
+  /// Changes the currently supported markets.
+  Future<void> setMarkets(
+    List<HistoricDocument> history,
+    List<FeedMarket> markets,
+  );
+
   /// Retrieves at most [maxDocuments] feed documents.
-  Future<List<DocumentWithActiveData>> getFeedDocuments(int maxDocuments);
+  Future<List<DocumentWithActiveData>> getFeedDocuments(
+    List<HistoricDocument> history,
+    int maxDocuments,
+  );
 
   /// Process the feedback about the user spending some time on a document.
   Future<void> timeSpent(TimeSpent timeSpent);
 
   /// Process the user's reaction to a document.
   Future<void> userReacted(UserReacted userReacted);
 }
+
+/// Passed to constructors/initializers of `Engine` implementing classe.
+class EngineInitializer with EquatableMixin {
```

Comment:
what's the purpose of this wrapper? the `Engine` effectively takes the same arguments and just unwraps this, rendering the wrapper somewhat pointless. and even though the `MockEngine` can take this optionally and store it in a field, this is never used there. is this wrapper important for a following pr?

Reply:
Mainly I don't want to repeat the same number of keyword arguments across various call places, and update them manually every time we add/change a field (like the new `aiConfig`).

Like its used by:

- `initializeEngine` in event_handler.dart
- `initialize` in `engine.dart` (ffi)
- the constructor of `MockEngine` (in the future) so that we can better test if the right data is passed in

Also maybe at some point make it a FFI type and pass this as a single boxed struct.